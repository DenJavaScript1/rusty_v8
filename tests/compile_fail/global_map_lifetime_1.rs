// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.
use rusty_v8 as v8;

pub fn main() {
  let mut isolate = v8::Isolate::new(mock());
  let scope = &mut v8::HandleScope::new(&mut isolate);
  let local = v8::Integer::new(scope, 42);
  let global = v8::Global::new(scope, local);
  let _ = global.map(|that| that);
}

fn mock<T>() -> T {
  unimplemented!()
}
