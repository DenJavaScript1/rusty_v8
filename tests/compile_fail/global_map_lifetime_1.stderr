error: lifetime may not live long enough
 --> $DIR/global_map_lifetime_1.rs:9:29
  |
9 |   let _ = global.map(|that| that);
  |                       ----- ^^^^ returning this value requires that `'1` must outlive `'2`
  |                       |   |
  |                       |   return type of closure is &'2 rusty_v8::Local<'_, rusty_v8::Integer>
  |                       has type `&'1 rusty_v8::Local<'_, rusty_v8::Integer>`

error: lifetime may not live long enough
 --> $DIR/global_map_lifetime_1.rs:9:29
  |
9 |   let _ = global.map(|that| that);
  |                       ----- ^^^^ returning this value requires that `'3` must outlive `'4`
  |                       |   |
  |                       |   return type of closure is &rusty_v8::Local<'4, rusty_v8::Integer>
  |                       has type `&rusty_v8::Local<'3, rusty_v8::Integer>`
